ADAL/src/ADEnrollmentGateway.m

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "ADEnrollmentGateway.h"
+#import "NSURL+ADExtensions.h"
+#import "ADAuthorityValidation.h"
+#import "ADAuthenticationError+Internal.h"
+
+// Keys for Intune Enrollment ID
+#define AD_INTUNE_ENROLLMENT_ID @"intune_app_protection_enrollment_id_V"
+#define AD_INTUNE_ENROLLMENT_ID_VERSION @"1"
+#define AD_INTUNE_ENROLLMENT_ID_KEY (AD_INTUNE_ENROLLMENT_ID AD_INTUNE_ENROLLMENT_ID_VERSION)
+
+// Keys for Intune Resource
+#define AD_INTUNE_RESOURCE_ID @"intune_mam_resource_V"
+#define AD_INTUNE_RESOURCE_ID_VERSION @"1"
+#define AD_INTUNE_RESOURCE_ID_KEY (AD_INTUNE_RESOURCE_ID AD_INTUNE_RESOURCE_ID_VERSION)
+
+NSString * const ENROLLMENT_ID_ARRAY = @"enrollment_ids";
```

Comment:
I talked to Olga last Friday, and she suggested that we could stick with the style `NSString *const`, as it is already used in our new code :)